src/components/views/avatars/DecoratedRoomAvatar.tsx

```diff
@@ -86,7 +86,9 @@ export default class DecoratedRoomAvatar extends React.PureComponent<IProps, ISt
         super(props);
 
         this.state = {
-            notificationState: RoomNotificationStateStore.instance.getRoomState(this.props.room),
+            notificationState: props.displayBadge
+                ? RoomNotificationStateStore.instance.getRoomState(this.props.room)
+                : undefined,
```

Comment:
This doesn't handle displayBadge changing during the lifespan of the component instance

Reply:
Reverted this optimisation, there's more implications i didn't think of you're right!